### PR TITLE
[policies] Add RedFlag OS policy

### PR DESCRIPTION
**Pull request by gogv**
---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [X] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

**Comment by gogv** [comment]
snap Build From Checkout Failing after 2m — Task Summary

**Comment by gogv** [comment]
I am building and publishing locally normally. What does this snap detect?

**Comment by arif-ali** [comment]
@gogv Thanks for your contribution

I have verified, that I see the same issue building the snap in my local environment, this is very likely an issue with snapcraft and not with your commit. I will get back to this issue at some point.

Furthermore, could you follow the contribution guideline at [1]

1. Can you put everything in english please, so that everyone can understand the code
2. Can you add The `Signed-off-by:` line too please to your commit
3. Once everything is approved you will need to squash your commits

[1] https://github.com/sosreport/sos/wiki/Contribution-Guidelines

**Comment by arif-ali** [comment]
@gogv once PR #3277 gets merged, please re-base your commits on that please

File: sos/policies/distros/redflag.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+# 版权所有（c）Yang Qing <yangqing@redflag-os.com>
+#
+# 此文件是sos项目的一部分：https://github.com/sosreport/sos
+#
+# 该受版权保护的材料可供任何希望使用的人使用，
+# 根据以下条款和条件对其进行修改、复制或重新分发
+# GNU通用公共许可证的第2版。
+#
+# 有关详细信息，请参阅源发行版中的LICENSE文件。
```
Comment (arif-ali) [inline sos/policies/distros/redflag.py:11]: Can you update this based on the guidelines, and can you put this in English please

File: sos/policies/distros/redflag.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        super(RedFlagPolicy, self).__init__(sysroot=sysroot, init=init,
+                                            probe_runtime=probe_runtime,
+                                            remote_exec=remote_exec)
+
+    @classmethod
+    def check(cls, remote=''):
+        if remote:
+            return cls.distro in remote
+
+        # 如果/etc/os-release丢失，则返回False
```
Comment (arif-ali) [inline sos/policies/distros/redflag.py:33]: same as above

File: sos/policies/distros/redflag.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        if not os.path.exists(OS_RELEASE):
+            return False
+
+        # 如果/etc/redflag-release丢失，则返回False
```
Comment (arif-ali) [inline sos/policies/distros/redflag.py:37]: same as above

File: sos/policies/distros/redflag.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        if not os.path.isfile('/etc/redflag-release'):
+            return False
+
+        # 检查/etc/os-release中的RedFlag
```
Comment (arif-ali) [inline sos/policies/distros/redflag.py:41]: same as above

**Review by arif-ali — COMMENTED** [review]
Can you update the PR based on my comments

* You don't need to add the commits from `main` branch, but you need to rebase to main, please see [1] as an example on how to do this (But you need for the snap PR to be merged first to finalise)
* Please follow the guidelines [2], and update the `rfedflag.py` using the License Header and Copyright


[1] https://stackoverflow.com/questions/54674971/github-fetch-and-rebase-fork-on-forked-master
[2] https://github.com/sosreport/sos/wiki/Contribution-Guidelines

File: sos/policies/distros/redflag.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+# 有关详细信息，请参阅源发行版中的LICENSE文件。
+
+from sos.policies.distros.redhat import RedHatPolicy, OS_RELEASE
+import os
+
+
+class RedFlagPolicy(RedHatPolicy):
+    distro = "RedFlag"
+    vendor = "RedFlag Software"
+    vendor_urls = [('Distribution Website', 'http://www.redflag-os.com')]
```
Comment (pmoravec) [inline sos/policies/distros/redflag.py:20]: My original point was a nitpick "can you provide rather a link to https pages, please?" but it seems the project runs (untrusted) http pages only..? (then this line would be fully OK).

**Comment by pmoravec** [comment]
As @arif-ali wrote: please use English in comments and rebase your branch first (to have PR with just one commit).

**Review by TurboTurtle — CHANGES_REQUESTED** [review]
The policy looks ok by itself, however you'll need to rebase and squash your fixup commits - this should ideally be contributed in a single commit.

**Comment by TurboTurtle** [comment]
Also, I'm a little concerned about the state of this distro.  A cursory Google search shows that places like [DistroWatch](https://distrowatch.com/table.php?distribution=redflag) has it marked as "discontinued", and several other communities have it noted that development stopped as far back as 2013, and that the commercial entity behind it went out of business in 2014.


So, before we move any further - what's the current status of the distribution, its community, and/or any commercial entity backing it?

**Comment by gogv** [comment]
In 2014, it was acquired by Beijing Hongqi Software and currently mainly provides services to Bank of China.

**Comment by gogv** [comment]
https://zh.wikipedia.org/wiki/%E7%B4%85%E6%97%97Linux

**Comment by arif-ali** [comment]
@gogv this PR has still got 22 commits, which is not correct; and has commits that are already in the `main` branch

We should be rabasing as per my comments 2 weeks ago; as per my comment on Friday and from @TurboTurtle this PR or any PR should only have your commit (or change), and ideally 1 commit

**Pull request closed without merging**